contracts/vaults/RibbonThetaVault.sol

```diff
@@ -412,11 +467,11 @@ contract RibbonThetaVault is DSMath, OptionsVaultStorage {
 
         _setNextOption(otokenAddress);
         _closeShort(oldOption);
-    }
 
-    function closeShort() external nonReentrant {
-        address oldOption = currentOption;
-        _closeShort(oldOption);
+        // After closing the short, if the options expire in-the-money
+        // vault pricePerShare would go down because vault's asset balance decreased.
+        // This ensures that the newly-minted shares do not take on the loss.
+        _mintPendingShares();
```

Comment:
when you say "newly minted shares" do you mean people who deposited in the current round? So you rebalance the price per share by inflating the price per share (by minting) so that the current round's depositors withdrawal amount will not take into account losses?

Reply:
Not sure what you mean by "rebalancing the price per share". The process of minting and burning doesn't actually affect the price per share.

Let me give you an example, when `balance = 100`, `shareSupply = 100`, pricePerShare will be `1`.

```js
pricePerShare = balance / shareSupply
```

 When you mint new shares, you don't actually update the pricePerShare value at all.

```js
// We mint
balance += 10
sharesMinted = pricePerShare * balance
shareSupply += sharesMinted
// pricePerShare remains the same!
```

Reply:
In the case of minting after closing the short, we can think of it as "let's use the pricePerShare right *after* the vault takes the loss".

Starting scenario:
```js
balance = 100
shareSupply = 100
pricePerShare = balance / shareSupply // pricePerShare = 1
```

After closing the short:
```js
closeShort() // causes balance to be -10, so 90
pricePerShare = balance / shareSupply // 90/100 = 0.9

mintShares() // mint shares using the new pricePerShare
```

Reply:
yep! makes sense